src/main/webapp/app/shared/profile-toggle/profile-toggle-link.directive.ts

```diff
@@ -0,0 +1,53 @@
+import { ChangeDetectorRef, Directive, HostBinding, Input, OnDestroy, OnInit, SkipSelf } from '@angular/core';
+import { ProfileToggle, ProfileToggleService } from 'app/shared/profile-toggle/profile-toggle.service';
+import { tap } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+
+@Directive({
+    selector: '[jhiProfileToggleLink]',
+})
+export class ProfileToggleLinkDirective implements OnInit, OnDestroy {
+    @Input('jhiProfileToggleLink') profile: ProfileToggle;
+    private profileActive = true;
+
+    private profileToggleActiveSubscription: Subscription;
+
+    constructor(@SkipSelf() protected cdRef: ChangeDetectorRef, private profileToggleService: ProfileToggleService) {}
+
+    /**
+     * Life cycle hook called by Angular to indicate that Angular is done creating the component
+     */
+    ngOnInit() {
+        // If no profile is set for the toggle, the directive does nothing.
+        if (this.profile) {
```

Comment:
question: Can this be `false`? Following the type it should never be `undefined` or `null`.

Reply:
I've added the `?` now to indicate that it could be `undefined`